channeldb/invoices.go

```diff
@@ -1263,70 +468,24 @@ func (d *DB) ScanInvoices(
 	}, reset)
 }
 
-// InvoiceQuery represents a query to the invoice database. The query allows a
-// caller to retrieve all invoices starting from a particular add index and
-// limit the number of results returned.
-type InvoiceQuery struct {
-	// IndexOffset is the offset within the add indices to start at. This
-	// can be used to start the response at a particular invoice.
-	IndexOffset uint64
-
-	// NumMaxInvoices is the maximum number of invoices that should be
-	// starting from the add index.
-	NumMaxInvoices uint64
-
-	// PendingOnly, if set, returns unsettled invoices starting from the
-	// add index.
-	PendingOnly bool
-
-	// Reversed, if set, indicates that the invoices returned should start
-	// from the IndexOffset and go backwards.
-	Reversed bool
-}
-
-// InvoiceSlice is the response to a invoice query. It includes the original
-// query, the set of invoices that match the query, and an integer which
-// represents the offset index of the last item in the set of returned invoices.
-// This integer allows callers to resume their query using this offset in the
-// event that the query's response exceeds the maximum number of returnable
-// invoices.
-type InvoiceSlice struct {
-	InvoiceQuery
-
-	// Invoices is the set of invoices that matched the query above.
-	Invoices []Invoice
-
-	// FirstIndexOffset is the index of the first element in the set of
-	// returned Invoices above. Callers can use this to resume their query
-	// in the event that the slice has too many events to fit into a single
-	// response.
-	FirstIndexOffset uint64
-
-	// LastIndexOffset is the index of the last element in the set of
-	// returned Invoices above. Callers can use this to resume their query
-	// in the event that the slice has too many events to fit into a single
-	// response.
-	LastIndexOffset uint64
-}
-
 // QueryInvoices allows a caller to query the invoice database for invoices
 // within the specified add index range.
-func (d *DB) QueryInvoices(q InvoiceQuery) (InvoiceSlice, error) {
-	var resp InvoiceSlice
+func (d *DB) QueryInvoices(q invPkg.InvoiceQuery) (invPkg.InvoiceSlice, error) {
```

Comment:
Hmm typically import names are just all lower case (so just `invpkg` here in this case). Why can't just `invoices` be used as well?

Reply:
Some files have functions with parameters/internal variables called `invoices`. For these cases I changed the import instead of the variables. I am happy to do it the other way, but it sad that we cannot use a proper name for a variable (`invoices`) because it's taken by the imported package.